src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,216 @@
+}
+
+type FileWidgetProps = WidgetProps & {
+	onChange: (params?: OnFileReadSuccessParams) => void;
+	onError?: (errors: string[]) => void;
+	value?: OnFileReadSuccessParams | string;
+};
+
+export const FileWidget = ({
+	onChange,
+	onError,
+	value,
+	...props
+}: FileWidgetProps) => {
+	const [isLoading, setIsLoading] = React.useState<boolean>(false);
+	const [loadingPercentage, setLoadingPercentage] = React.useState<number>(0);
+	const [errors, setErrors] = React.useState<string[]>([]);
+
+	const accept =
+		props.uiSchema?.['ui:options']?.accept ??
+		props.uiSchema?.value?.['ui:options']?.accept;
+	const maxSize =
+		props.uiSchema?.['ui:options']?.maxSize ??
+		props.uiSchema?.value?.['ui:options']?.maxSize;
+
+	const dataUrl = (value: OnFileReadSuccessParams | string) => {
+		return typeof value === 'string' ? value : value.dataUrl;
+	};
+
+	const isImage = (value: OnFileReadSuccessParams | string) => {
+		return dataUrl(value).startsWith('data:image/');
+	};
+
+	const onDrop = React.useCallback(
+		((acceptedFiles, fileRejections) => {
+			setErrors([]);
+
+			if (fileRejections != null && fileRejections.length > 0) {
+				const errs = fileRejections[0].errors.map(
+					({ message }) => `${fileRejections[0].file.name}: ${message}`,
+				);
+				setErrors(errs);
+				onChange();
+				return onError?.(errs);
+			}
+
+			acceptedFiles.forEach((file) => {
+				const reader = new FileReader();
+				reader.onabort = () => console.log('file reading was aborted');
+				reader.onerror = () => {
+					const err = [`Failed to upload ${file.name}`];
+					onChange();
+					setErrors(err);
+					onError?.(err);
+				};
+				reader.onloadstart = () => setIsLoading(true);
+				reader.onprogress = (event) => {
+					setLoadingPercentage(Math.floor((100 * event.loaded) / event.total));
+				};
+				reader.onload = () => {
+					if (file != null && reader.result != null) {
+						setIsLoading(false);
+						onChange({
+							dataUrl: reader.result as string,
+							uploadedFile: file,
+						});
+					}
+				};
+
+				reader.readAsDataURL(file);
+			});
+		}) as NonNullable<DropzoneOptions['onDrop']>,
+		[],
+	);
+
+	const { getRootProps, getInputProps, isDragActive } = useDropzone({
+		onDrop,
+		maxSize,
+		accept,
+		multiple: false,
+		noDrag: false,
+		disabled: isLoading,
+	});
+
+	return (
+		<>
+			{props.schema.title && <InputLabel>{props.schema.title}</InputLabel>}
+			<Box
+				sx={{
+					border: (theme) =>
+						`dashed 2px ${
+							errors.length !== 0
+								? '#F19B9E'
+								: isDragActive
+								? theme.palette.primary.main
+								: 'rgba(0, 0, 0, 0.23)'
+						}`,
+					borderRadius: '2px',
+					display: 'flex',
+					flexDirection: 'column',
+					alignItems: 'center',
+					justifyContent: 'center',
+				}}
+				style={{ minHeight: '150px' }}
+				flexDirection="column"
+				alignItems="center"
+				justifyContent="center"
+				{...getRootProps()}
+				dir={undefined}
+			>
+				<input {...getInputProps()} />
```

Comment:
Not sure if this should be replaced with a `<TextField>` and have the input props passed in through the appropriate props on that (called `InputProps` I think) cc @JSReds @drskullster

Reply:
Maybe I can just use MUI <Input /> ? https://mui.com/material-ui/api/input/

Reply:
We're using TextField for all inputs, but that is in part a design decision. Would defer to Andrea and Jon in case they have another reason for preferring one or the other in this situation, but I believe the bottom line is that regardless of which input we use it will be hidden, right?